packages/frontend/src/ClientWithStateManager.ts

```diff
@@ -5,29 +5,31 @@ import { type IRelayEndpoint, RelayClient } from "@massmarket/client";
 import { LevelStore } from "@massmarket/store/level";
 
 export class ClientWithStateManager {
-  public stateManager;
-  public relayClient;
+  public stateManager: Database;
+  public relayClient: RelayClient;
 
   constructor(
-    public readonly relayEndpoint: IRelayEndpoint,
-    walletClient: WalletClient,
-    account: Account,
+    private readonly relayEndpoint: IRelayEndpoint,
+    private readonly walletClient: WalletClient,
+    private readonly account: Account,
     public readonly shopId: bigint,
-  ) {
-    const store = new LevelStore();
-    this.stateManager = new Database(
-      {
-        store,
-        objectId: shopId,
-      },
-    );
+  ) {}
+
+  async initialize() {
```

Comment:
```suggestion
  async open() {
```

to not confuse things, personally, i'd prefer to just mirror the name of the wrapped thing